cmd/seal.go

```diff
@@ -0,0 +1,113 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func getFlagString(name string, cmd *cobra.Command) string {
+	value, err := cmd.Flags().GetString(name)
+	if value == "" || err != nil {
+		return ""
+	}
+	return fmt.Sprintf("--%s %s", name, value)
+}
+
+func getFlagFilePath(name string, cmd *cobra.Command) string {
+	value, err := cmd.Flags().GetString(name)
+	if value == "" || err != nil {
+		return ""
+	}
+	filePath, err := os.Getwd()
+	if err == nil {
+		filePath = fmt.Sprintf("%s/%s", filePath, value)
+	} else {
+		filePath = value
+	}
+	return fmt.Sprintf("--%s %s", name, filePath)
+}
+
+func getFlagFilePathSlice(name string, cmd *cobra.Command) string {
+	argList, err := cmd.Flags().GetStringSlice(name)
+	if err != nil || len(argList) == 0 {
+		return ""
+	}
+	result := ""
+	filePath, err := os.Getwd()
+	if err == nil {
+		filePath = fmt.Sprintf("%s/", filePath)
+	} else {
+		filePath = ""
+	}
+	for _, value := range argList {
+		value = fmt.Sprintf("--%s %s/%s", name, filePath, value)
+		result = fmt.Sprintf("%s %s", result, value)
+	}
+	return fmt.Sprintf("--%s %s", name, result)
+}
+
+func getFlagBool(name string, cmd *cobra.Command) string {
+	value, _ := cmd.Flags().GetBool(name)
+	if !value {
+		return ""
+	}
+	return fmt.Sprintf("--%s", name)
+}
+
+var Seal = &cobra.Command{
+	Use:   "seal",
+	Short: "Seal a secret using sealed-secrets",
+	Args:  cobra.MinimumNArgs(0),
+	Run: func(cmd *cobra.Command, args []string) {
+		flags := []string{
+			getFlagFilePath("cert", cmd),
+			getFlagString("controller-namespace", cmd),
+			getFlagString("controller-name", cmd),
+			getFlagString("format", cmd),
+			getFlagFilePath("sealed-secret-file", cmd),
+			getFlagFilePath("secret-file", cmd),
+			getFlagBool("fetch-cert", cmd),
+			getFlagBool("allow-empty-data", cmd),
+			getFlagBool("validate", cmd),
+			getFlagFilePath("merge-into", cmd),
+			getFlagBool("raw", cmd),
+			getFlagBool("re-encrypt", cmd),
+			getFlagString("name", cmd),
+			getFlagFilePathSlice("from-file", cmd),
+			getFlagBool("recovery-unseal", cmd),
+			getFlagFilePathSlice("recovery-private-key", cmd),
+		}
+
+		p := getPlatform(cmd)
+		kubeseal := p.GetBinary("kubeseal")
+		flagString := strings.Join(flags, " ")
+		argString := strings.Join(args, " ")
+		if err := kubeseal(flagString + argString); err != nil {
+			log.Fatalf("failed to run kubeseal: %v", err)
+		}
+	},
+}
+
+func init() {
+	Seal.Flags().String("cert", "", "Certificate/public key file/URL to use for encryption")
+	Seal.Flags().String("controller-namespace", metav1.NamespaceSystem, "Namespace of sealed-secrets-controller")
+	Seal.Flags().String("controller-name", "sealed-secrets-controller", "Name of sealed-secrets controller.")
```

Comment:
So these fields should be based on platform.SealedSecrets, the interface for the user would be:

`karina seal secret.yaml -c karina.yaml > sealed.yaml`
and 
`karina unseal sealed.yaml -c karina.yaml > secret.yaml`